app-backend/tile/src/main/scala/image/GlobalSummary.scala

```diff
@@ -0,0 +1,97 @@
+package com.azavea.rf.tile.image
+
+import com.azavea.rf.tile._
+import com.azavea.rf.tile.tool.TileSources
+import com.azavea.rf.datamodel.{Tool, ToolRun, User, MosaicDefinition}
+import com.azavea.rf.tool.eval._
+import com.azavea.rf.tool.ast._
+import com.azavea.rf.tool.params._
+import com.azavea.rf.tool.ast.MapAlgebraAST
+import com.azavea.rf.common._
+import com.azavea.rf.common.cache._
+import com.azavea.rf.common.cache.kryo.KryoMemcachedClient
+import com.azavea.rf.database.Database
+import com.azavea.rf.database.tables._
+
+import io.circe._
+import io.circe.syntax._
+import geotrellis.raster._
+import geotrellis.raster.render._
+import geotrellis.raster.histogram._
+import geotrellis.raster.io._
+import geotrellis.vector.io._
+import geotrellis.spark.io._
+import geotrellis.spark._
+import geotrellis.proj4._
+import geotrellis.spark.io.s3.{S3InputFormat, S3AttributeStore, S3CollectionLayerReader, S3ValueReader}
+import com.github.blemale.scaffeine.{Scaffeine, Cache => ScaffeineCache}
+import geotrellis.vector.Extent
+import com.typesafe.scalalogging.LazyLogging
+import spray.json.DefaultJsonProtocol._
+import cats._
+import cats.data._
+import cats.implicits._
+
+import java.security.InvalidParameterException
+import java.util.UUID
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.util._
+
+
+object GlobalSummary extends LazyLogging {
+
+  /** Get the [[RasterExtent]] which describes the meaningful subset of a layer from metadata */
+  private def getDefinedRasterExtent(md: TileLayerMetadata[_]): RasterExtent = {
+    val re = RasterExtent(md.layout.extent,
+      md.layout.tileLayout.totalCols.toInt,
+      md.layout.tileLayout.totalRows.toInt
+    )
+    val gb = re.gridBoundsFor(md.extent)
+    re.rasterExtentFor(gb).toRasterExtent
+  }
+
+  /** Get the minimum zoom level for a single scene from which a histogram can be constructed without
+    *  losing too much information (too much being defined by the 'size' threshold)
+    */
+  def minAcceptableSceneZoom(sceneId: UUID, store: AttributeStore, size: Int = 512): Option[(Extent, Int)] = {
+    def startZoom(zoom: Int): Option[(Extent, Int)] = {
+      val currentId = LayerId(sceneId.toString, zoom)
+      val metadata = Try { store.readMetadata[TileLayerMetadata[SpatialKey]](currentId) }.toOption
+      metadata.flatMap { meta =>
+        val re = getDefinedRasterExtent(meta)
+        logger.debug(s"Data Extent: ${meta.extent.reproject(WebMercator, LatLng).toGeoJson()}")
+        logger.debug(s"$currentId has (${re.cols},${re.rows}) pixels")
+        if (re.cols >= size || re.rows >= size) Some((meta.extent, currentId.zoom))
+        else startZoom(zoom + 1).orElse(Some(meta.extent, currentId.zoom))
+      }
+    }
+    startZoom(1)
+  }
```

Comment:
`O(n)` search for the right zoom should be good, considering we generally know the zoom cap. Probably not worth it to binsearch on the zoom layers instead.